cluster-autoscaler/cloudprovider/linode/linode_cloud_config.go

```diff
@@ -0,0 +1,164 @@
+	if err := gcfg.ReadInto(&gcfgCloudConfig, config); err != nil {
+		return nil, err
+	}
+
+	// get the clusterID
+	clusterID, err := strconv.Atoi(gcfgCloudConfig.Global.ClusterID)
+	if err != nil {
+		return nil, fmt.Errorf("LKE Cluster ID %q is not a number: %v", gcfgCloudConfig.Global.ClusterID, err)
+	}
+
+	// get the linode token
+	token := gcfgCloudConfig.Global.Token
+	if len(gcfgCloudConfig.Global.Token) == 0 {
+		return nil, fmt.Errorf("linode token not present in global section")
+	}
+
+	// get the list of LKE pools that must not be imported
+	excludedPoolIDs := make(map[int]bool)
+	for _, excludedPoolIDStr := range gcfgCloudConfig.Global.ExcludedPoolIDs {
+		excludedPoolID, err := strconv.Atoi(excludedPoolIDStr)
+		if err != nil {
+			return nil, fmt.Errorf("excluded pool id %q is not a number: %v", excludedPoolIDStr, err)
+		}
+		excludedPoolIDs[excludedPoolID] = true
+	}
+
+	// get the default min and max size as defined in the global section of the config file
+	defaultMinSize, defaultMaxSize, err := getSizeLimits(
+		gcfgCloudConfig.Global.DefaultMinSize,
+		gcfgCloudConfig.Global.DefaultMaxSize,
+		defaultMinSizePerLinodeType,
+		defaultMaxSizePerLinodeType)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get default size values in global section: %v", err)
+	}
+
+	// get the specific configuration of a node group
+	nodeGroupCfg := make(map[string]*nodeGroupConfig)
+	for nodeType, gcfgNodeGroup := range gcfgCloudConfig.NodeGroups {
+		minSize, maxSize, err := getSizeLimits(gcfgNodeGroup.MinSize, gcfgNodeGroup.MaxSize, defaultMinSize, defaultMaxSize)
+		if err != nil {
+			return nil, fmt.Errorf("cannot get size values for node group %q: %v", nodeType, err)
+		}
+		ngc := &nodeGroupConfig{
+			maxSize: maxSize,
+			minSize: minSize,
+		}
+		nodeGroupCfg[nodeType] = ngc
+	}
+
+	return &linodeConfig{
+		clusterID:       clusterID,
+		token:           token,
+		defaultMinSize:  defaultMinSize,
+		defaultMaxSize:  defaultMaxSize,
+		excludedPoolIDs: excludedPoolIDs,
+		nodeGroupCfg:    nodeGroupCfg,
+	}, nil
+}
+
+// getSizeLimits takes the max, min size of a node group as strings (empty if no values are provided)
+// and default sizes, validates them an returns them as integer, or an error if such occurred
+func getSizeLimits(minStr string, maxStr string, defaultMin int, defaultMax int) (int, int, error) {
+	var err error
+	min := defaultMin
+	if len(minStr) != 0 {
+		min, err = strconv.Atoi(minStr)
+		if err != nil {
+			return 0, 0, fmt.Errorf("could not parse min size for node group: %v", err)
+		}
+	}
+	if min < 1 {
+		return 0, 0, fmt.Errorf("min size for node group cannot be < 1")
+	}
+	max := defaultMax
+	if len(maxStr) != 0 {
+		max, err = strconv.Atoi(maxStr)
+		if err != nil {
+			return 0, 0, fmt.Errorf("could not parse max size for node group: %v", err)
+		}
+	}
+	if min >= max {
+		return 0, 0, fmt.Errorf("min size for a node group must be strictly less than its max size (got min: %d, max: %d)",
```

Comment:
Why can't min be equal to max?

Reply:
I guessed that if min size == max size then it didn't make sense to use the cluster-autoscaler for this node group, since the size is fixed then, but yes, you are right, if the user want to configure it that way why not, I will change this.

Reply:
I've updated the PR allowing min == max.